dask_yarn/cli.py

```diff
@@ -440,42 +442,41 @@ async def run():
     ),
 )
 def worker(nthreads=None, memory_limit=None):  # pragma: nocover
-    enable_proctitle_on_current()
-    enable_proctitle_on_children()
-
     if memory_limit is None:
         memory_limit = int(skein.properties.container_resources.memory * 2 ** 20)
     if nthreads is None:
         nthreads = skein.properties.container_resources.vcores
 
+    worker_options = dict(memory_limit=memory_limit, nthreads=nthreads)
     app_client = skein.ApplicationClient.from_current()
 
     scheduler = app_client.kv.wait("dask.scheduler").decode()
-
-    loop = IOLoop.current()
-
-    worker = Nanny(
+    worker_class = os.environ.get("worker_class", "dask.distributed.Nanny")
+    options = os.environ.get("worker_options", str(base64.b64encode(b"{}")))
+
+    # options come in a string
+    # encode string as bytes
+    # decode base64
+    # load with json
+    options = base64.b64decode(options.encode())
+    worker_options.update(json.loads(options))
+    print(worker_options)
+    command = [
+        sys.executable,
+        "-m",
+        "distributed.cli.dask_spec",
         scheduler,
-        loop=loop,
-        memory_limit=memory_limit,
-        worker_port=0,
-        nthreads=nthreads,
-        name=skein.properties.container_id,
-    )
-
-    async def cleanup():
-        await worker.close(timeout=2)
-
-    install_signal_handlers(loop, cleanup=cleanup)
```

Comment:
I'm curious why the handlers are being removed here.

Reply:
I thought I should remove the cleanup function since we are no longer creating a worker/Nanny directly.  What would a good cleanup function here be ?  kill the process ?

Reply:
I think just `install_signal_handlers(loop)` allows for a more graceful closing of the IOLoop if a SIGTERM/SIGINT is sent.

Reply:
Thanks!  Resolved in https://github.com/dask/dask-yarn/pull/140/commits/53582520032cfda9ae74e522ac35ed373b2b099e